core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -1063,9 +1048,7 @@ public void openExternalUrl(Intent intent) {
     if (intent.resolveActivity(getActivity().getPackageManager()) != null) {
```

Comment:
we have 2 identical methods now. I think maybe the original intent was to share this functionality?

You can create a class ExternalLinkOpener that is scoped to the ActivityComponent and inject it where you need it

Reply:
hmmm, maybe after researching what `isExternalLinkPopup` is actually doing

Reply:
ooh, great idea!